tests/auth/test_auth_endpoint.py

```diff
@@ -0,0 +1,23 @@
+"""Testing auth endpoint."""
+
+import jwt
+
+from src.config import Config
+from src.users.models import User
+
+
+def test_post_auth_endpoint_with_nonexistent_user_returns_400_with_correct_message(client, db_empty):
+    result = client.post("/api/auth", data={"username": "test", "password": "testtest"})
+
+    assert result.status_code == 400 and b"couldn't find user with credentials provided" in result.data
+
+
+def test_post_auth_endpoint_with_existing_user_returns_correct_body(client, db_with_one_user):
+    session = db_with_one_user
+    user = session.query(User).first()
+    expected_hash = str.encode(jwt.encode({"user_id": user.id}, Config.SECRET_KEY, algorithm="HS256"))
+
+    result = client.post("/api/auth", data={"username": "test", "password": "testtest"})
+
+    assert result.status_code == 200
+    assert expected_hash in result.data
```

Comment:
Since it's better to check the whole response body whenever it's possible, let's change this assertion to something like:
```python
assert result.json == jwt.encode({"user_id": 1}, Config.SECRET_KEY, algorithm="HS256")
```